pkg/entities/set.go

```diff
@@ -58,22 +59,27 @@ type set struct {
 	isDecoding bool
 }
 
-func NewSet(setType ContentType, templateID uint16, isDecoding bool) Set {
-	set := &set{
+func NewSet(setType ContentType, isDecoding bool) Set {
+	return &set{
 		buffer:     &bytes.Buffer{},
 		setType:    setType,
 		records:    make([]Record, 0),
 		isDecoding: isDecoding,
 	}
-	if !isDecoding {
+}
+
+func (s *set) PrepareSet(templateID uint16) error {
```

Comment:
Is it possible that we `PrepareSet` after `ResetSet`? I think this is a valid use case for us to re-use the set. If so, we can consider moving the `setType` parameter to `PrepareSet` otherwise the `setType` will be `Undefined`

Reply:
Agreed it is good to have these checks. Yes, we can reuse the set for both data and template types instead of allocating again. 
Added those checks.

Reply:
I think the problem is still there -- if we call `ResetSet` and then want to use `PrepareSet`, the setType will always be `Undefined`.

Reply:
And it's better to add another check [here](https://github.com/vmware/go-ipfix/blob/5993fb33828e98810e63f3aa8e1cab451783624e/pkg/exporter/process.go#L179) so that `Undefined` setType will not go through this function.